code/go/0chain.net/blobbercore/reference/referencepath.go

```diff
@@ -22,6 +22,71 @@ func GetReferencePath(ctx context.Context, allocationID, path string) (*Ref, err
 	return GetReferencePathFromPaths(ctx, allocationID, []string{path})
 }
 
+func GetReferencePath2(ctx context.Context, allocationID, path string) (*Ref, error) {
+	return GetReferenceForHashCalculationFromPaths(ctx, allocationID, []string{path})
+}
+
+// GetReferenceForHashCalculationFromPaths validate and build full dir tree from db, and CalculateHash and return root Ref without saving in DB
+func GetReferenceForHashCalculationFromPaths(ctx context.Context, allocationID string, paths []string) (*Ref, error) {
+	var refs []Ref
+	db := datastore.GetStore().GetTransaction(ctx) //"created_at", "updated_at",
+	db = db.Select("id", "allocation_id", "type", "name", "path",
+		"parent_path", "size", "hash", "path_hash", "content_hash", "merkle_root",
+		"actual_file_size", "actual_file_hash", "attributes", "chunk_size",
+		"lookup_hash", "thumbnail_hash")
+	pathsAdded := make(map[string]bool)
+	for _, path := range paths {
+		path = strings.TrimSuffix(path, "/")
+		if _, ok := pathsAdded[path]; !ok {
+			db = db.Where(Ref{ParentPath: path, AllocationID: allocationID})
+			pathsAdded[path] = true
+		}
+		depth := len(GetSubDirsFromPath(path)) + 1
+		curPath := filepath.Dir(path)
```

Comment:
This `curPath` prevails from previous code. I kind of don't like this logic. We could simply split path with `/` and get parent path by joining elements.
Something like:
```
splittedPaths := strings.Split("path", "/")
for i:=0;i<len(splittedPaths);i++{
    newPath := strings.Join(splittedPaths[0:i], "/")
}
```

Reply:
this update may trigger errors and wrong paths if the paths include "." as parent_path. and give wrong paths for the DB request. for now I may keep it until I figure out why the paths are not similar.